datafusion/expr/src/expr.rs

```diff
@@ -392,11 +392,34 @@ impl Unnest {
 }
 
 /// Alias expression
-#[derive(Clone, PartialEq, Eq, PartialOrd, Hash, Debug)]
+#[derive(Clone, PartialEq, Eq, Debug)]
 pub struct Alias {
     pub expr: Box<Expr>,
     pub relation: Option<TableReference>,
     pub name: String,
+    pub metadata: Option<std::collections::HashMap<String, String>>,
```

Comment:
Aliases may be lost during expression rewriting. eg .
https://github.com/apache/datafusion/blob/a176aacc92a5b0272690bb48e7084ff5603ef7ad/datafusion/optimizer/src/analyzer/type_coercion.rs#L1018

According to the description of the PR, I think only columns/fields have metadata, not all expressions.
`Field` already has customizable [metadata](https://github.com/apache/arrow-rs/blob/f5138fcb10040e2428f2dd540177f5f224d0281e/arrow-schema/src/field.rs#L53); is it possible to use it to meet the requirements🤔?